balancer-js/src/modules/pools/queries/params_builder.ts

```diff
@@ -35,13 +36,14 @@ export class ParamsBuilder implements PoolQueries.ParamsBuilder {
     );
     const assets = [...this.pool.tokensList];
 
-    // Remove BPT token from amounts
-    if (bptIndex && bptIndex > -1) {
-      maxAmountsIn.splice(bptIndex, 1);
+    let maxInWithoutBpt = [...maxAmountsIn];
+    // Remove BPT token from amounts for user data
+    if (bptIndex > -1) {
+      maxInWithoutBpt = removeItem(maxAmountsIn, bptIndex);
```

Comment:
I might be missing something, but i don't think the change is needed. Can there be a join with BPT?

Reply:
The original test was including a value for the BPT (and was failing for new pool with BPT at 0 index) so I thought that was how the param worked. As you suggest in other comment I think a refactor would make things clearer and should remove that.